Install in an isolated namespace

I'm attempting to install sealed secrets helm chart into a namespace where my user is "admin" at the role level not the cluster level.  I was hoping to install sealed secrets in this namespace to isolate the users from needing access to other namespaces/secrets.

Am I being a numbskull here?

```---
kind: Role
apiVersion: rbac.authorization.k8s.io/v1beta1
metadata:
  name: {{ item.namespace }}-manager
  namespace: {{ item.namespace }}
rules:
- apiGroups:
  - "*"
  resources:
  - "*"
  verbs:
  - "*"
```

```
---
kind: RoleBinding
apiVersion: rbac.authorization.k8s.io/v1beta1
metadata:
  name: {{ item.namespace }}-binding
  namespace: {{ item.namespace }}
subjects:
{% for item in item.users | default([]) %}
- kind: User
  name: {{ item }}
  apiGroup: rbac.authorization.k8s.io
{% endfor -%}
roleRef:
  kind: Role
  name: {{ item.namespace }}-manager
  apiGroup: rbac.authorization.k8s.io
```

I guess this just isn't the way to go about it.  I am running the service in `kube-system` which is the intended way to do this.

The user I have has the access above, but needs access  to services/proxy in cluster role.

```
kubeseal <mysecret.json >mysealedsecret.json
panic: Error fetching certificate: services "http:sealed-secrets-controller:" is forbidden: User "xxx" cannot get resource "services/proxy" in API group "" in the namespace "kube-system"
```

--- a/cmd/controller/server.go
+++ b/cmd/controller/server.go
@@ -26,6 +26,9 @@ type certProvider func() []*x509.Certificate
 type secretChecker func([]byte) (bool, error)
 type secretRotator func([]byte) ([]byte, error)
 
+// httpserver starts an HTTP that exposes core functionality like serving the public key
+// or secret rotation and validation. This endpoint is designed to be accessible by
+// all users of a given cluster. It must not leak any secret material.
 func httpserver(cp certProvider, sc secretChecker, sr secretRotator) {
 	httpRateLimiter := rateLimter()
 

--- a/controller.jsonnet
+++ b/controller.jsonnet
@@ -38,6 +38,26 @@ controller {
     ],
   },
 
+  serviceProxierRole: kube.Role('sealed-secrets-service-proxier') + $.namespace {
+    rules: [
+      {
+        apiGroups: [
+          '',
+        ],
+        resources: [
+          'services/proxy',
+        ],
+        resourceNames: [
+          'http:sealed-secrets-controller:',  // kubeseal uses net.JoinSchemeNamePort when crafting proxy subresource URLs
+          'sealed-secrets-controller',  // but often services are referred by name only, let's not make it unnecessarily cryptic
+        ],
+        verbs: [
+          'get',
+        ],
+      },
+    ],
+  },
+
   unsealerBinding: kube.ClusterRoleBinding('sealed-secrets-controller') {
     roleRef_: $.unsealerRole,
     subjects_+: [$.account],
@@ -46,6 +66,13 @@ controller {
   unsealKeyBinding: kube.RoleBinding('sealed-secrets-controller') + $.namespace {
     roleRef_: $.unsealKeyRole,
     subjects_+: [$.account],
+  },
+
+  serviceProxierBinding: kube.RoleBinding('sealed-secrets-service-proxier') + $.namespace {
+    roleRef_: $.serviceProxierRole,
+    // kube.libsonnet assumes object here have a namespace, but system groups don't
+    // thus are not supposed to use the magic "_" here.
+    subjects+: [kube.Group('system:authenticated')],
   },
 
   controller+: {

